sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CollectionQuerySimplification.cs

```diff
@@ -46,23 +46,36 @@ public class CollectionQuerySimplification : SonarDiagnosticAnalyzer
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
 
-        private static readonly ISet<string> MethodNamesWithPredicate = ImmutableHashSet.Create(
+        private static readonly ISet<string> MethodNamesWithPredicate = new HashSet<string>

```

Comment:
Converting ImmutableHasSet to HashSet (several times)

Reply:
I think it is worth doing a separate commit changing all the `ImmutableHashSet` instances to `HashSet` to avoid doing this so often.